lib/src/list/list_builder.dart

```diff
@@ -48,6 +44,15 @@ class ListBuilder<E> {
     updates(this);
   }
 
+  /// Replaces all elements with elements from an [Iterable].
+  void replace(Iterable iterable) {
+    if (iterable is BuiltList<E>) {
+      _replaceFromBuiltList(iterable);
+    } else {
+      _replaceFromSafeList(new List<E>.from(iterable));
```

Comment:
This check is already done in checked mode by List, but I agree we should check it.

I'll file a separate issue to improve testing around type checking on instantiation.